### PR TITLE
Feature/web sockets

### PR DESCRIPTION
**Pull request by DikenMaharjan**

### CODE REVIEW

File: src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        threadInteractionSyncListener.didAddThread(
+            threadDTO = thread
+        ).also {
+            serverProxy.localServer.createOrUpdateThread(
+                threads = listOf(thread)
+            )
+        }
+    }
+
+    private suspend fun ThreadAssets.notifyNewAssetInThread() {
```
Comment (gennaro-safehill) [inline src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt:65]: Is this for both photo messages and non?

File: src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        coroutineScope {
+            val interactionsSummary = runCatchingPreservingCancellationException {
+                serverProxy.topLevelInteractionsSummary()
+            }
+            interactionsSummary.onSuccess { interactionsSummaryDTO ->
+                val allThreads = interactionsSummaryDTO.summaryByThreadId.map {
+                    it.value.thread
+                }
+                val localThreads = serverProxy.localServer.listThreads()
+                deleteUnwantedThreadsLocally(
```
Comment (gennaro-safehill) [inline src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt]: Didn't we say we would rename this?

File: src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+interface InteractionSyncListener {
+    suspend fun didUpdateThreadsList(threadDTOs: List<ConversationThreadOutputDTO>)
+
+    suspend fun didAddThread(threadDTO: ConversationThreadOutputDTO)
+
+    suspend fun didReceiveTextMessages(messageDtos: List<MessageOutputDTO>, threadId: String)
+
+    suspend fun didReceivePhotoMessages(
```
Comment (gennaro-safehill) [inline src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt:144]: How do you distinguish between photo messages and assets shared with the people in the thread?

**Review by gennaro-safehill — APPROVED** [review]

File: src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt (hunk 1 above)
Comment (DikenMaharjan) [reply src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt:65]: No.. its only for photo messages.

Comment (DikenMaharjan) [reply src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt:65]: Non photo messages are given by RemoteDownloadOperation for now.

File: src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt (hunk 3 above)
Comment (DikenMaharjan) [reply src/main/kotlin/com/safehill/kclient/tasks/syncing/ThreadInteractionSync.kt:144]: As of now, only the photo messages are notified by this method and  whereas assets shared outside of thread is notified to us by the RemoteDownloadOperation.

**Pull request merged**